halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/GenerateCommand.java

```diff
@@ -1,4 +1,37 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package com.netflix.spinnaker.halyard.cli.command.v1;
 
-public class GenerateCommand {
+import com.netflix.spinnaker.halyard.cli.services.v1.Daemon;
+import com.netflix.spinnaker.halyard.cli.services.v1.DaemonService;
+import lombok.AccessLevel;
+import lombok.Getter;
+
+public class GenerateCommand extends NestableCommand {
+  @Getter(AccessLevel.PUBLIC)
+  private String commandName = "generate";
+
+  @Getter(AccessLevel.PUBLIC)
+  private String description = "Generate the full Spinnaker config for you current deployment.";
+
+  @Override
+  protected void executeThis() {
+    DaemonService service = Daemon.getService();
+    String currentDeployment = service.getCurrentDeployment();
+    service.generateDeployment(currentDeployment, "", false);
```

Comment:
Well in some way we need the user to express what version of Spinnaker they want to deploy, and I also want to be ready for the scenario where there is a breaking config change that needs to be addressed by halyard or the user. I don't see a way around these two things without writing the version of spinnaker to deploy in the config. 

I think once you start having the user think in config deltas, we are making the upgrade path too complicated - ideally the 90% case will be a config transform where we rename/replace fields on behalf of the user. Other config changes happen in the background and are invisible to the user when the upgrade to a new version.